sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -459,7 +459,8 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] with AliasHelpe
       case alias @ Alias(_: AttributeReference, _) =>
         (alias.exprId, Literal.create(null, alias.dataType))
       case alias @ Alias(l: Literal, _) =>
-        (alias.exprId, l.copy(value = null))
+        // SPARK-43156: copy value to get the correct value when extracting in constructLeftJoins
+        (alias.exprId, l.copy(value = l.value))
```

Comment:
what does `l.copy(value = l.value)` do? seems nothing?

Reply:
My fault, just want to create new `Literal` instance with seem value.